spot_driver/launch/spot_driver.launch.py

```diff
@@ -51,12 +139,28 @@ def launch_setup(context: LaunchContext, ld: LaunchDescription) -> None:
     rviz_config_file = LaunchConfiguration("rviz_config_file").perform(context)
     spot_name = LaunchConfiguration("spot_name").perform(context)
     tf_prefix = LaunchConfiguration("tf_prefix").perform(context)
-    use_depth_registered_nodelets = LaunchConfiguration("use_depth_registered_nodelets")
+    depth_registered_mode_config = LaunchConfiguration("depth_registered_mode")
+    publish_point_clouds_config = LaunchConfiguration("publish_point_clouds")
 
     pkg_share = FindPackageShare("spot_description").find("spot_description")
 
+    depth_registered_mode_string = depth_registered_mode_config.perform(context).lower()
+    if depth_registered_mode_string == "from_spot":
+        depth_registered_mode = DepthRegisteredMode.FROM_SPOT
+    elif depth_registered_mode_string == "from_nodelets":
+        depth_registered_mode = DepthRegisteredMode.FROM_NODELETS
+    elif depth_registered_mode_string == "disable":
+        depth_registered_mode = DepthRegisteredMode.DISABLE
+    else:
+        depth_registered_mode = DepthRegisteredMode.DISABLE
```

Comment:
I added an error message and early failure if an invalid option is provided. I agree that it's preferable to communicate the issue to the user rather than failing silently.